contributors/design-proposals/pod-resource-management.md

```diff
@@ -1,114 +1,397 @@
 
-We add a flag `qos-memory-overcommitment` to kubelet which would allow users to configure the percentage of memory overcommitment on the node. We have the default as 100, so by default we allow complete overcommitment on the node and let the BE pod use as much memory as it wants, and not reserve any resources for the G and Bu pods. As expected if there is an OOM in such a case we first kill the BE pods before the G and Bu pods.
-On the other hand if a user wants to ensure very predictable tail latency for latency-sensitive servers he would need to set qos-memory-overcommitment to a really low value(preferrably 0). In this case memory resources would be reserved for the G and Bu pods and BE pods would be able to use only the left over memory resource.
+The `ROOT` cgroup sandbox is used to parent all pod sandboxes that are in
+the Guaranteed QoS class.  By definition, pods in this class have cpu and 
+memory limits specified that are equivalent to their requests so the pod 
+level cgroup sandbox confines resource consumption without the need of an
+additional cgroup sandbox for the tier.
 
-Examples in the next section.
+When the `kubelet` launches, it will ensure a `Burstable` cgroup sandbox
+and a `BestEffort` cgroup sandbox exist as children of `ROOT`.  These cgroup
+sandboxes will parent pod level cgroups in those associated QoS classes.
 
-### Proposed cgroup hierarchy:
+The `kubelet` highly prioritizes resource utilization, and thus
+allows BestEffort and Burstable pods to potentially consume as many
+resources that are presently available on the node.
 
-For the initial implementation we will only support limits for cpu and memory resources.
+For compressible resources, this prioritization scheme has little impact.
+For example, CPU time is proportioned dynamically when there is contention
+using CFS shares that ensure minimum requests are satisfied.
 
-#### QoS classes
+For incompressible resources, this prioritization scheme can inhibit the
+ability of a pod to have its requests satisfied.  For example, a Guaranteed
+pods memory request may not be satisfied if there are active BestEffort
+pods consuming all available memory.
 
-A pod can belong to one of the following 3 QoS classes: Guaranteed, Burstable, and BestEffort, in decreasing order of priority.
+The `kubelet` will support a flag `--qos-reserve-limits` that takes a
+set of percentages per compressible resource that controls how the QoS
+cgroup sandbox attempts to reserve requested resources across QoS classes
+based on the set of pods active on the node.  The flag will accept values
+in a range from 0-100%, where a value of 0 instructs the `kubelet` to
+attempt no reservation, and a value of 100 will instruct the `kubelet`
+to attempt to reserve the sum of requested resource across all pods
+on the node.  How the `kubelet` achieves this desired state is resource
+specific.  The default value per compressible resource if not specified
+is for no reservation to occur.
```

Comment:
Should to default be 100% reservation? I think this is what Vish was thinking.  Since 0% is the effective value if you don't use the flag.  The flag without a value should do something different than not having flag at all.